packages/flutter/test/material/icon_button_test.dart

```diff
@@ -3032,6 +3032,309 @@ void main() {
         ..rect(color: const Color(0xFF00FF00)), // IconButton overlay.
     );
   });
+
+  testWidgets('Material3 - IconButton variants hovered & onLongPressed', (WidgetTester tester) async {
+    late bool onHovered;
+    bool onLongPressed = false;
+
+    void onLongPress() {
+      onLongPressed = true;
+    }
+
+    void onHover(bool hover) {
+      onHovered = hover;
+    }
+
+    // IconButton
+    await tester.pumpWidget(build(onLongPress: onLongPress, onHover: onHover));
+
+    final Finder iconButton = find.widgetWithIcon(IconButton, Icons.favorite);
+    final Offset iconButtonOffset = tester.getCenter(iconButton);
+    final TestGesture gesture = await tester.createGesture(kind: PointerDeviceKind.mouse);
+    await gesture.addPointer();
+
+    await gesture.moveTo(iconButtonOffset);
+    await tester.pumpAndSettle();
```

Comment:
A general rule for Flutter widget tests is to try as much as possible to avoid `tester.pumpAndSettle()` as it can hide issues.

Solutions to try in order:
- replace with tester.pump()
- replace with several tester.pump() and comment why there are needed (might be needed in case something is postponed to the next frame).
- replace with tester.pump(duration) and comment why this duration (or use an explicit name for the duration variable). This is needed when there are animations for instance.